task-runner/task_runner/file_manager.py

```diff
@@ -77,18 +77,25 @@ def upload_output(
         task_id: str,
         task_dir_remote: str,
         local_path: str,
+        stream_zip: bool = True,
     ):
         del task_dir_remote  # unused
 
+        if stream_zip:
+            data = files.get_zip_generator(local_path)
+            size = data.total_bytes
```

Comment:
This cannot be done here. Before yielding data from the generator the value of `total_bytes` will be 0. This value must be fetched after the upload request.

Reply:
I moved the size calculation to after the upload request